File: integration-test/java/quarkus-app/src/main/java/com/qualityminds/lazyval/integration/boundary/persistence/cassandra/CassandraOrderRepository.java
package com.qualityminds.lazyval.integration.boundary.persistence.cassandra;

import com.datastax.oss.quarkus.runtime.api.session.QuarkusCqlSession;
import com.qualityminds.lazyval.integration.domain.Order;
import com.qualityminds.lazyval.integration.domain.OrderRepository;
import com.qualityminds.lazyval.integration.shared.Isbn;
import io.smallrye.common.annotation.Identifier;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;

import java.util.Comparator;
import java.util.List;
import java.util.UUID;

@ApplicationScoped
@Identifier("cassandra")
public class CassandraOrderRepository implements OrderRepository {

    private final CassandraOrderDao dao;
    private final QuarkusCqlSession session;
    private final CassandraMapper mapper;

    @Inject
    public CassandraOrderRepository(CassandraOrderDao dao, QuarkusCqlSession session, CassandraMapper mapper) {
        this.dao = dao;
        this.session = session;
        this.mapper = mapper;
    }

    @Override
    public void save(Order order) {
        dao.update(mapper.toDB(order));
    }


    @Override
    public List<Order> findAll() {
        return mapper.toDomain(dao.findAll().all()).stream()
                .sorted(Comparator.comparing(order -> order.isbn().value()))
                .toList();
    }

    @Override
    public Order getById(UUID id) {
        return mapper.toDomain(dao.getById(id));
    }

    @Override
    public List<Order> findByIsbn(Isbn isbn) {
        return mapper.toDomain(dao.findByIsbn(isbn).all()).stream()
                .sorted(Comparator.comparing(order -> order.isbn().value()))
                .toList();
    }
}

File: integration-test/kotlin/quarkus-app/src/main/kotlin/com/qualityminds/lazyval/integration/boundary/persistence/cassandra/CassandraOrderRepository.kt
package com.qualityminds.lazyval.integration.boundary.persistence.cassandra

import com.datastax.oss.quarkus.runtime.api.session.QuarkusCqlSession
import com.qualityminds.lazyval.integration.domain.Order
import com.qualityminds.lazyval.integration.domain.OrderRepository
import com.qualityminds.lazyval.integration.shared.Isbn
import io.smallrye.common.annotation.Identifier
import jakarta.enterprise.context.ApplicationScoped
import jakarta.inject.Inject
import java.util.UUID

@ApplicationScoped
@Identifier("cassandra")
class CassandraOrderRepository @Inject constructor(
    private val dao: CassandraOrderDao,
    private val session: QuarkusCqlSession,
    private val mapper: CassandraMapper
) : OrderRepository {

    override fun save(order: Order) {
        dao.update(mapper.toDB(order))
    }

    override fun findAll(): List<Order> =
        mapper.toDomain(dao.findAll().all()).sortedBy { it.isbn.value }

    override fun getById(id: UUID): Order? =
        dao.getById(id)?.let { mapper.toDomain(it) }

    override fun findByIsbn(isbn: Isbn): List<Order> =
        mapper.toDomain(dao.findByIsbn(isbn).all()).sortedBy { it.isbn.value }
}
